sopht_mpi/numeric/eulerian_grid_ops/stencil_ops_2d/outplane_field_curl_mpi_2d.py

```diff
@@ -0,0 +1,116 @@
+"""MPI-supported kernels for computing curl of outplane field in 2D."""
+from sopht.numeric.eulerian_grid_ops.stencil_ops_2d import (
+    gen_outplane_field_curl_pyst_kernel_2d,
+)
+from sopht_mpi.utils.mpi_utils import check_valid_ghost_size_and_kernel_support
+
+
+def gen_outplane_field_curl_pyst_mpi_kernel_2d(
+    real_t, mpi_construct, ghost_exchange_communicator
+):
+    """MPI-supported 2D outplane field curl kernel generator."""
+    outplane_field_curl_pyst_kernel_2d = gen_outplane_field_curl_pyst_kernel_2d(
+        real_t=real_t
+    )
+    kernel_support = 1
+    # define this here so that ghost size and kernel support is checked during
+    # generation phase itself
+    gen_outplane_field_curl_pyst_mpi_kernel_2d.kernel_support = kernel_support
+    check_valid_ghost_size_and_kernel_support(
+        ghost_size=ghost_exchange_communicator.ghost_size,
+        kernel_support=gen_outplane_field_curl_pyst_mpi_kernel_2d.kernel_support,
+    )
+
+    def outplane_field_curl_pyst_mpi_kernel_2d(curl, field, prefactor):
+        """MPI-supporteed outplane field curl in 2D.
+        Computes curl of outplane 2D vector field (field)
+        into vector 2D inplane field (curl_x, curl_y).
+        Used for psi ---> velocity
+        Assumes curl field is (2, n, n).
+        """
+        # define kernel support for kernel
+        outplane_field_curl_pyst_mpi_kernel_2d.kernel_support = (
+            gen_outplane_field_curl_pyst_mpi_kernel_2d.kernel_support
+        )
+        # define variable for use later
+        ghost_size = ghost_exchange_communicator.ghost_size
+        # begin ghost comm.
+        ghost_exchange_communicator.exchange_init(curl[0], mpi_construct)
+        ghost_exchange_communicator.exchange_init(curl[1], mpi_construct)
```

Comment:
I feel we don't need this. Can you try commenting on this and see if the tests pass?

Reply:
Good catch. That is the output and shouldn't need to be ghosted anyway. Thanks. Fixed in [586ec40](https://github.com/fankiat/sopht-mpi/pull/24/commits/586ec403844b1c39b5f7227ab49bc69ce7a61688)